examples/dd: add XCOPY support

ddiss opened the pull request:
EXTENDED COPY can be triggered with the new --xcopy/-x parameter. When
invoked, (--max) EXTENDED COPY requests are dispatched in parallel, with
each request attempting to copy (--blocks) from source to destination.

Submitting for early feedback - I should probably split out the populate_X helpers into a library that can be shared with the test-tool code, as currently a lot of this is duplicate.

sahlberg commented:
Very Nice.

Agree. I think the populate_ functions should go into the main library,
as well a new function iscsi_xcopy_async().

As this is new functionality, that does not block me merging the current
patch as it.

Your call, I can do either.

* merge as is now and you, me or someone else adds a proper
iscsi_xcopy_async() call afterwards
* another spin of the patch where you do these changes?

(
I don't think TGTD has XCOPY support yet, so I  might try adding it as this
will be a very nice tool to
test and verify with. This is very helpful for that.
)

On Mon, Jan 2, 2017 at 3:20 AM, David Disseldorp <notifications@github.com>
wrote:

> EXTENDED COPY can be triggered with the new --xcopy/-x parameter. When
> invoked, (--max) EXTENDED COPY requests are dispatched in parallel, with
> each request attempting to copy (--blocks) from source to destination.
>
> Submitting for early feedback - I should probably split out the populate_X
> helpers into a library that can be shared with the test-tool code, as
> currently a lot of this is duplicate.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/sahlberg/libiscsi/pull/230
> Commit Summary
>
>    - examples/dd: add XCOPY support
>
> File Changes
>
>    - *M* examples/iscsi-dd.c
>    <https://github.com/sahlberg/libiscsi/pull/230/files#diff-0> (307)
>
> Patch Links:
>
>    - https://github.com/sahlberg/libiscsi/pull/230.patch
>    - https://github.com/sahlberg/libiscsi/pull/230.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sahlberg/libiscsi/pull/230>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAeNkMFcyYi-izF_KDFdbI5eO5ZaRvYHks5rON1_gaJpZM4LY9FQ>
> .
>

ddiss commented:
Thanks for the feedback Ronnie.
I think some duplication of the populate_X functions will still be needed, as the tests are poking into obsolete / reserved fields. I'll whip up a quick iscsi_xcopy_async() change and resubmit with that, the common populate_X functions could then be done at a later date.

ddiss commented:
Changes since previous submission:
- add libiscsi RECEIVE COPY RESULTS and EXTENDED COPY async/sync helper functions
- validate blocks per I/O and max in flight parameters
  + check blocks per I/O value doesn't exceed XCOPY limits
- split out duplicate readcap code

sahlberg commented:
Awesome, thanks!

plieven commented:
dont we need to bump the api version?

> Am 05.01.2017 um 03:14 schrieb Ronnie Sahlberg <notifications@github.com>:
> 
> Awesome, thanks!
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>
